### PR TITLE
Validate channel in application resource

### PR DESCRIPTION
**Pull request by Aflynn50**
## Description
When specifying an application resource, the channel parameter specifies the channel on charmhub from which to fetch the charm. Right now, this is parsed by `charm.Parse` and anything it accepts is valid.

However, users have been running into issues where problems are caused due to the fact that their plan only specifies the risk, and not the track (#445). The default track is resolved by juju and errors like this are caused by the fact that juju has resolved the track and it is not present in the plan.

Rather than fixing the particular issues caused by these inconstancy, it makes more sense to for the users to specify themselves the actual track that they need, rather than relying on juju to resolve it. By requiring this, we make the juju terraform plans more repeatable (the default track could change between applications of a plan, meaning you can get different states for the same plan). It also removes this class of bugs.

It is still possible to omit a channel entirely, this may be desirable for non production plans where a user wants to just quickly test something and not have to bother going to charmhub to look up the default track.

Fixes: 
#445
## Type of change
- Change existing resource
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Environment

- Juju controller version: 

- Terraform version: 

## QA steps

Manual QA steps should be done to test this PR.

```tf
terraform {
  required_providers {
    juju = {
      version = "~> 0.11.0"
      source  = "juju/juju"
    }
  }
}

provider "juju" {}

resource "juju_model" "development" {
  name = "development"

  cloud {
    name   = "lxd"
  }
}

resource "juju_application" "wordpress" {
  name = "wordpress"

  model = juju_model.development.name

  charm {
    name = "ubuntu"
    base = "ubuntu@22.04"
    channel = "latest"
  }
}
```
On `terraform plan` this returns the error:
```
╷
│ Error: Invalid Channel
│ 
│   with juju_application.wordpress,
│   on plan.tf line 28, in resource "juju_application" "wordpress":
│   28:     channel = "latest"
│ 
│ String must conform to track/risk or track/risk/branch, e.g. latest/stable
```
When the channel is changed to "stable/latest" then `terraform plan & terraform apply` works to deploy the model
## Additional notes

JUJU-5858

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -67,9 +67,7 @@ juju-unit-test:
 .PHONY: envtestlxd
 envtestlxd:
 ## envtestlxd: Under development - Include env var and run unit tests against lxd
-	JUJU_CONTROLLER_ADDRESSES=${CONTROLLER_ADDRESSES} \
-	JUJU_USERNAME=${USERNAME} JUJU_PASSWORD=${PASSWORD} \
-	JUJU_CA_CERT=${CA_CERT} TF_ACC=1 TEST_CLOUD=lxd go test ./... -v $(TESTARGS) -timeout 120m
+	JUJU_CONTROLLER_ADDRESSES=${CONTROLLER_ADDRESSES} JUJU_USERNAME=${USERNAME} JUJU_PASSWORD=${PASSWORD} JUJU_CA_CERT=${CA_CERT} TF_ACC=1 TEST_CLOUD=lxd go test ./... -v $(TESTARGS) -timeout 120m
```
Comment (cderici) [inline Makefile:70]: Looks like your editor consolidated these into one line, I'd keep it on separate lines for readability.

**Review by cderici — APPROVED** [review]
This LGTM, QA went well. Tried it with a bunch of different channel values (including ""), and the validation catches it during the plan as expected, nice work 👍 

I think we should also have some unit tests for this, see [internal application tests](https://github.com/juju/terraform-provider-juju/blob/main/internal/juju/applications_test.go).

File: Makefile (hunk 1 above)
Comment (Aflynn50) [reply Makefile]: Changed back

**Comment by Aflynn50** [comment]
> This LGTM, QA went well. Tried it with a bunch of different channel values (including ""), and the validation catches it during the plan as expected, nice work 👍
> 
> I think we should also have some unit tests for this, see [internal application tests](https://github.com/juju/terraform-provider-juju/blob/main/internal/juju/applications_test.go).

I've added some unit tests for the validator

**Pull request merged**